src/serviceFacades/vice/admin.js

```diff
@@ -51,23 +53,56 @@ export const downloadInputFiles = ({ analysisID }) =>
         method: "POST",
     });
 
-export const getUserJobLimit = ({ username }) =>
-    callApi({
+export const getUserJobLimit = ({ username }) => {
+    const existsPromise = userExists({ username });
+    const jobLimitPromise = callApi({
         endpoint: `/api/admin/vice/concurrent-job-limits/${username}`,
         method: "GET",
     });
 
-export const setUserJobLimit = ({ username, newLimit }) =>
-    callApi({
+    return existsPromise.then((doesExist) => {
+        if (!doesExist) {
+            throw new Error(`User ${username} does not exist`);
+        }
+        return jobLimitPromise;
+    });
+};
+
+export const setUserJobLimit = ({ username, newLimit }) => {
+    const existsPromise = userExists({ username });
+    const setJobLimitPromise = callApi({
         endpoint: `/api/admin/vice/concurrent-job-limits/${username}`,
         method: "PUT",
         body: {
             concurrent_jobs: parseInt(newLimit, 10),
         },
     });
 
-export default () =>
+    return existsPromise.then((doesExist) => {
+        if (!doesExist) {
+            throw new Error(`User ${username} does not exist`);
+        }
+        return setJobLimitPromise;
+    });
+};
+
+export const userExists = ({ username }) => {
+    const suffix = `@${constants.IPLANT}.org`;
+
+    if (!username.endsWith(suffix)) {
+        username = `${username}${suffix}`;
+    }
+
+    return callApi({
+        endpoint: `/api/workspaces?username=${username}`,
+        method: "GET",
+    }).then((data) => data.workspaces.length > 0);
+};
```

Comment:
I think we can still use react-query for this logic: https://react-query.tanstack.com/docs/guides/queries#dependent-queries